test/typings/should_not_compile/tsconfig.json

```diff
@@ -0,0 +1,11 @@
+{
+  "compilerOptions": {
+    "module": "commonjs",
+    "allowUnreachableCode": true
```

Comment:
This only works in TypeScript 1.8, so we won't be able to test backwards compatibility. Still, having sanity-check tests are definitely better than nothing, so it's worth merging in - I can open a ticket for definition test coverage.